File: Baekjoon/구현/10773.py
# https://www.acmicpc.net/problem/10773

k = int(input())
stack = []

for i in range(k): 
    num = int(input())
    if(num == 0):
        stack.pop()
    else:
        stack.append(num)
print(sum(stack))

File: Baekjoon/구현/10773.cpp
// https://www.acmicpc.net/problem/10773

#include <iostream>
#include <stack>
using namespace std;

int main() {
    int k;
    cin >> k;

    stack<int> s;
    for (int i = 0; i < k; i++) {
        int n;
        cin >> n;

        if (n == 0) {
            s.pop();
        } else {
            s.push(n);
        }
    }

    int sum = 0;
    while (!s.empty()) {
        sum += s.top();
        s.pop();
    }

    cout << sum << '\n';

    return 0;
}